### PR TITLE
Fix skipped tests

### LINKED ISSUE #19: Fix tests
[Issuehunt badges](https://issuehunt.io/r/sindresorhus/timed-out/issues/19)

A couple of tests were failing, so they are skipped. Would be nice to figure out why they are failing now.

- https://github.com/sindresorhus/timed-out/blob/9cc4ef15197e9017dd84e2639bbb7a6e09f3adf7/test.js#L21-L34
- https://github.com/sindresorhus/timed-out/blob/9cc4ef15197e9017dd84e2639bbb7a6e09f3adf7/test.js#L194-L231

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Backers (Total: $40.00)

- [issuehunt issuehunt](https://issuehunt.io/u/issuehunt) ($40.00)
### Submitted pull Requests
- [#20 Fix tests, remove unnecessary exit-clause](https://issuehunt.io/r/sindresorhus/timed-out/pull/20)
- [#24 Fix skipped tests](https://issuehunt.io/r/sindresorhus/timed-out/pull/24)
---

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/timed-out/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
---
IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### PR DESCRIPTION
**Pull request by danedavid**
Fixes previously skipped tests. Fixes #19.

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Referenced issues

This pull request has been submitted to:
- [#19: Fix tests](https://issuehunt.io/repos/26973848/issues/19)
---

IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### CODE REVIEW

**Comment by danedavid** [comment]
For node > v8, `socket._idleTimeout` returns `undefined`.  Thus using `socket.timeout` whenever available, using conditional code. Can revamp if there is a better option. Can change the check to node version check as well. Or skip the test for Node v8.

File: test.js [hunk 1]
```diff
@@ -224,7 +224,7 @@ describe('when connection is established', () => {
 
 		request.on('socket', socket_ => {
 			socket_.once('connect', () => {
-				assert.equal(socket_.timeout, 100);
+				assert.equal(isNaN(socket.timeout) ? socket._idleTimeout : socket.timeout, 100);
```
Comment (sindresorhus) [inline test.js]: Use a helper function named `socketTimeout` so you don't have to repeat the logic all over and use `Number.isNaN`.

Comment (danedavid) [reply test.js]: `Number.isNaN()` won't work if the passed value is not of type `number`. I have used `isNaN` to check for `undefined`, so replacing with `in` operator which makes more sense in this case.

Comment (sindresorhus) [reply test.js]: That's a better check, indeed.

**Pull request merged**